types/index.d.ts

```diff
@@ -280,10 +280,7 @@ declare module 'found' {
      */
     getComponent?: (
       match: RouteMatch,
-    ) =>
-      | React.ComponentType<any>
-      | Promise<React.ComponentType<any>>
-      | Promise<JSX.Element>;
```

Comment:
i am really confused how having an extra type in a union can make dtslint fail. but i do think this type is wrong

Reply:
i don't understand dtslint AT ALL. it also always uses `tsc@next`

Reply:
i mean this is a real error. i just have no idea how this caught it:

```
yarn run v1.13.0
$ dtslint types
Error: /Users/jjia/vendor/found/types/example-app/index.tsx:34:11
ERROR: 34:11  expect  TypeScript@next compile error:
Type '() => Promise<ComponentClass<any, any> | FunctionComponent<any> | Element | (() => Element)>' is not assignable to type '(match: RouteMatch) => ComponentClass<any, any> | FunctionComponent<any> | Promise<ComponentType<any>> | Promise<Element>'.
  Type 'Promise<ComponentClass<any, any> | FunctionComponent<any> | Element | (() => Element)>' is not assignable to type 'ComponentClass<any, any> | FunctionComponfix: Fix Route getComponent typing
ent<any> | Promise<ComponentType<any>> | Promise<Element>'.
    Type 'Promise<ComponentClass<any, any> | FunctionComponent<any> | Element | (() => Element)>' is not assignable to type 'Promise<ComponentType<any>>'.
      Type 'ComponentClass<any, any> | FunctionComponent<any> | Element | (() => Element)' is not assignable to type 'ComponentType<any>'.
        Type 'Element' is not assignable to type 'ComponentType<any>'.
          Type 'Element' is not assignable to type 'FunctionComponent<any>'.
            Type 'Element' provides no match for the signature '(props: any, context?: any): ReactElement<any> | null'.
```